packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -549,6 +549,73 @@ class _AnimatedPaddingState extends AnimatedWidgetBaseState<AnimatedPadding> {
   }
 }
 
+/// Animated version of [Align] which automatically transitions the child's
+/// position over a given duration whenever the given [alignment] changes.
+///
+/// See also:
+///
+///  * [AnimatedContainer], which can transition more values at once.
+class AnimatedAlign extends ImplicitlyAnimatedWidget {
+  /// Creates a widget that positions its child by an alignment that animates
+  /// implicitly.
+  ///
+  /// The [alignment], [curve], and [duration] arguments must not be null.
+  const AnimatedAlign({
+    Key key,
+    @required this.alignment,
+    this.child,
+    Curve curve: Curves.linear,
+    @required Duration duration,
+  }) : assert(alignment != null),
+       super(key: key, curve: curve, duration: duration);
+
+  /// How to align the child.
+  ///
+  /// The x and y values of the alignment control the horizontal and vertical
+  /// alignment, respectively. An x value of -1.0 means that the left edge of
```

Comment:
This feels like a duplicate for the Alignment documentation.
Shouldn't we prefer to keep a single copy of a documentation, and refer to it from relevant places instead of copying?
(this way it's less likely that we change something and update only a subset of the doc comments that are talking about it...)

Reply:
(Summarising a discussion we had in person:)

The risk of docs getting out of date because they are duplicated is definitely an issue.
In theory we would fix this to some extent using dartdoc macros, though this isn't a panacea (sometimes the docs are slightly different each time, which means they can't use a macro, but would still go out of date; also, it's unfortunate for people who are reading the docs from the source).
In practice (based on our experience in usability studies) we want the docs to be immediately useful when people read them, without them having to chase a series of links to get to the information. So the duplication ends up being somewhat necessary. This is also why we sometimes have docs that paraphrase each other.

We'll have to be careful to keep the docs up to date. In theory we should change APIs rarely so it shouldn't be a huge time sink.